test/e2e/quick_start.go

```diff
@@ -147,7 +165,6 @@ func QuickStartSpec(ctx context.Context, inputGetter func() QuickStartSpecInput)
 				KubernetesVersion:        input.E2EConfig.GetVariable(KubernetesVersion),
 				ControlPlaneMachineCount: controlPlaneMachineCount,
 				WorkerMachineCount:       workerMachineCount,
```

Comment:
This now ignores the passed in input.ClusterctlVariables

We need something like this
```go
		variables := map[string]string{}
		for k, v := range input.ClusterctlVariables {
			variables[k] = v
		}
		if input.DeployClusterClassInSeparateNamespace {
			variables["CLUSTER_CLASS_NAMESPACE"] = clusterClassNamespace.Name
			By("Creating a cluster referencing a ClusterClass from another namespace")
		}
```